wazimap_ng/cache.py

```diff
@@ -65,55 +71,86 @@ def last_modified_point_updated(request, profile_id, profile_category_id=None, t
 
     return last_modified(request, profile_id, key)
 
+
 ########### Signals #################
 def update_profile_cache(profile):
     logger.info(f"Updating profile cache: {profile}")
     key = profile_key % profile.id
     cache.set(key, datetime.now())
 
+
 def update_point_cache(category):
     profile = category.profile
     key1 = location_key % (profile.id, category.id)
 
     logger.debug(f"Set cache key (category): {key1}")
     cache.set(key1, datetime.now())
 
-@receiver(post_save, sender=ProfileIndicator)
+
+@receiver(post_save, sender=profile_models.ProfileIndicator)
 def profile_indicator_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
-@receiver(post_save, sender=ProfileHighlight)
+
+@receiver(post_save, sender=profile_models.ProfileHighlight)
 def profile_highlight_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
-@receiver(post_save, sender=IndicatorCategory)
+
+@receiver(post_save, sender=profile_models.IndicatorCategory)
 def profile_category_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
-@receiver(post_save, sender=IndicatorSubcategory)
+
+@receiver(post_save, sender=profile_models.IndicatorSubcategory)
 def profile_subcategory_updated(sender, instance, **kwargs):
     update_profile_cache(instance.category.profile)
 
-@receiver(post_save, sender=ProfileKeyMetrics)
+
+@receiver(post_save, sender=profile_models.ProfileKeyMetrics)
 def profile_keymetrics_updated(sender, instance, **kwargs):
     update_profile_cache(instance.profile)
 
-@receiver(post_save, sender=Profile)
+
+@receiver(post_save, sender=profile_models.Profile)
 def profile_indicator_updated(sender, instance, **kwargs):
     update_profile_cache(instance)
 
-@receiver(post_save, sender=Location)
+
+@receiver(post_save, sender=points_models.Location)
 def point_updated_location(sender, instance, **kwargs):
     update_point_cache(instance.category)
 
-@receiver(post_save, sender=Category)
+
+@receiver(post_save, sender=points_models.Category)
 def point_updated_category(sender, instance, **kwargs):
     update_point_cache(instance)
 
+
+@receiver(post_save, sender=profile_models.Indicator)
+def indicator_updated(sender, instance, **kwargs):
+    indicator_obj = instance
+    all_profile_indicators = profile_models.ProfileIndicator.objects.filter(
```

Comment:
What if we consolidate those lookups into just one lookup using `Q` from django?

https://docs.djangoproject.com/en/3.1/topics/db/queries/#complex-lookups-with-q-objects

There is a nice talk from djangoCon on this topic: https://www.youtube.com/watch?v=p9Bgr5VOISo

Reply:
Doing it this way the code is much more readable. 
And we are having 4 different tables so we have to use the reverse related query which is sometimes daunting. If you insist I can give it a try but definitely it won't be a clean way to do it. 
Generally I follow this principle 
Code readability > optimisation (unless we have a bottleneck)

Reply:
I agree with your readability should be valued more than optimization. but not sure this applies here. 

```
from django.db.models import Q

profiles_to_invalidate_cache = Profile.objects.fliter(
  Q(indicators__indicator_id=id) 
  | Q(profilekeymetric_set__variable_id=id) 
  | Q(profilehighlight_set__indicator_id=id)
)
```

I don't this is less readable but we can have this discussion at a later point. 
I don't want to halt the merging of this too much.